app/src/test/java/com/android/unio/model/event/EventViewModelTest.kt

```diff
@@ -156,4 +170,13 @@ class EventViewModelTest {
     eventViewModel.selectEvent(testEvents[0].uid)
     assertEquals(testEvents[0], eventViewModel.selectedEvent.value)
   }
+
+  @Test
+  fun testAddEventUserPicture() {
+    `when`(eventUserPictureRepositoryFirestore.getNewUid()).thenReturn("1")
+    val picture =
+        EventUserPicture("0", "http://real-image.com", User.emptyFirestoreReferenceElement(), 0)
+    eventViewModel.addEventUserPicture(inputStream, testEvents[0], picture)
+    verify(eventUserPictureRepositoryFirestore).addEventUserPicture(any(), any(), any())
```

Comment:
This is minor but you should verify the correct `EventUserPicture` was used in the `addEventUserPicture ` function call, instead of `any()`

Reply:
I cannot make the check on the EventUserPicture because `eventViewModel.addEventUserPicture` modifies its uid and image fields. But I'm going to make the check on the nb of likes and on the user as those should be the same :)

Reply:
It is harder than I thought to verify the value of certain attributes without checking that the whole objects are equal :/ I will leave like this for now as I lost too much time on this

Reply:
> It is harder than I thought to verify the value of certain attributes without checking that the whole objects are equal :/ I will leave like this for now as I lost too much time on this

Yeah I think it's fine. I understand it's not easy because you would have to mock multiple other things in the view model for it.